src/main/java/software/amazon/nio/spi/s3/S3SeekableByteChannel.java

```diff
@@ -7,61 +7,58 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import software.amazon.nio.spi.s3.config.S3NioSpiConfiguration;
-import software.amazon.nio.spi.s3.util.TimeOutUtils;
 import software.amazon.awssdk.services.s3.S3AsyncClient;
 import software.amazon.awssdk.services.s3.model.HeadObjectResponse;
+import software.amazon.nio.spi.s3.config.S3NioSpiConfiguration;
+import software.amazon.nio.spi.s3.util.TimeOutUtils;
 
 import java.io.IOException;
 import java.nio.ByteBuffer;
-import java.nio.channels.ClosedChannelException;
-import java.nio.channels.ReadableByteChannel;
-import java.nio.channels.SeekableByteChannel;
-import java.nio.channels.WritableByteChannel;
+import java.nio.channels.*;
+import java.nio.file.OpenOption;
 import java.nio.file.StandardOpenOption;
+import java.util.Collections;
+import java.util.Set;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
 public class S3SeekableByteChannel implements SeekableByteChannel {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(S3SeekableByteChannel.class);
+
     private long position;
     private final S3AsyncClient s3Client;
     private final S3Path path;
     private final ReadableByteChannel readDelegate;
-    //private final WritableByteChannel writeDelegate; // placeholder for when we implement writes.
+    private final S3WritableByteChannel writeDelegate;
 
+    private boolean closed;
     private long size = -1L;
 
-    private final Logger logger = LoggerFactory.getLogger(this.getClass().getName());
+    protected S3SeekableByteChannel(S3Path s3Path, S3AsyncClient s3Client) throws IOException {
+        this(s3Path, s3Client, Collections.emptySet());
+    }
 
-    protected S3SeekableByteChannel(S3Path s3Path, S3AsyncClient s3Client, long startAt) throws IOException{
```

Comment:
Technically this would change the signature of a method that is visible to any subclass someone may have made. Can you use an overloaded method instead and perhaps deprecate this one?